src/app/boards/components/board-item/add-column/add-column.component.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { ColumnService } from 'src/app/boards/services/column.service';
+import { BoardsService } from 'src/app/boards/services/boards.service';
+import { Column, NewColumnData } from 'src/app/models/models';
+import { ModalService } from 'src/app/services/modal.service';
+import { SharedService } from 'src/app/shared/shared.service';
+
+@Component({
+  selector: 'app-add-column',
+  templateUrl: './add-column.component.html',
+  styleUrls: ['./add-column.component.scss'],
+})
+export class AddColumnComponent implements OnInit, OnDestroy {
+  formData!: FormGroup;
+  boardId?: string | null = '';
+  board?: string | null = '';
+  column?: Column;
+  columns: Column[] = [];
+  columnData?: Column | null;
+  private paramMapSubscription: Subscription | undefined;
+
+  constructor(
+    private columnService: ColumnService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private boardsService: BoardsService,
+    private modalService: ModalService,
+    private sharedService: SharedService
+  ) {}
+
+  ngOnInit(): void {
+    this.formData = new FormGroup({
+      title: new FormControl<string>('', [
+        Validators.required,
+        Validators.minLength(6),
+      ]),
+    });
+
+    this.paramMapSubscription = this.route.paramMap.subscribe((params) => {
```

Comment:
Every time visiting this page will be created a new subscription. This leads to performance issue. So you should pay attention to closing them. It might be unsubscribing ( in the onDestroy method that you are using) or approaches from the article https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f. In the projects' life, we use take* more often than others.